api/RS.API/RS.Service/Logic/OpeningManagerService.cs

```diff
@@ -145,23 +147,31 @@ public IResult GetOpeningsCorrespondingToLoggedUser(Guid userId)
             };
             try
             {
-                var allOpenings = _openingRepository.GetOpeningsCorrespondingToLoggedUser(userId);
-                var openingIds = allOpenings.Select(x => x.OpeningId).Distinct().ToList();
-                var approvalTransactions = _approvalRepository.GetAllApprovalTransactions(openingIds).ToList();
-                result.Body = allOpenings.Select(opening =>
-                {
-                    var approvalTransaction = approvalTransactions.FirstOrDefault(x => x.EntityId == opening.OpeningId);
-
-                    var openingViewModel = new OpeningViewModel();
-                    MapPrimaryandSecondarySkills(openingViewModel, opening);
+                var allOpenings = _openingRepository.GetOpeningsCorrespondingToLoggedUser(searchAndSortModel.UserId);
 
-                    if (approvalTransaction != null)
+                List<OpeningViewModel> openingModelList = new List<OpeningViewModel>();
+                if (searchAndSortModel.Property != null)
+                {
+                    if (searchAndSortModel.Direction == 1)
                     {
-                        openingViewModel.IsApproved = approvalTransaction.IsApproved;
-                        openingViewModel.Status = approvalTransaction.ApprovalAction.ApprovalActionName;
+                        openingModelList = allOpenings.OrderBy(x => x.GetType().GetProperty(searchAndSortModel.Property).GetValue(x, null)).ToList();
```

Comment:
why searching sorting and filtering implemented on fetched result set ?
Business operations are only performed on result set.

Reply:
Done